### ISSUE TITLE
Camvid Segnet  unlabelled problem

### ISSUE THREAD

**Issue by andyhahaha** [meetshah1995/pytorch-semseg#58]
I use this code to train segnet on camvid, but the results can't reach the paper performance.
I found that you didn't use median-frequency balancing to training.
Moreover, unlabelled class is '11', not '<0', so you didn't ignore unlabelled pixel at training and validate.
Your segnet will output unlabelled class that is weird. You should change ```n_class=12 ``` to ```n_class=11```.

**Comment by andyhahaha** [comment]
I follow the split from this repo.
https://github.com/alexgkendall/SegNet-Tutorial
This is the repo from segnet author alex kendall.

**Comment by d-li14** [comment]
@andyhahaha Thanks a lot for reminding me, and I got the split from SegNet repo. But there exists some conflicts with the [CamVid official site](http://mi.eng.cam.ac.uk/research/projects/VideoRec/CamVid/) as far as i know:
* it claims 32 classes and here we can only see 11
* the [label-color mapping](http://mi.eng.cam.ac.uk/research/projects/VideoRec/CamVid/data/label_colors.txt) is also not so matching with the `label_colous` list in the `camvid_loader.py` for lack of _Roadmarking_ and _Pavement_

**Comment by d-li14** [comment]
@andyhahaha Furthermore, as for your original question, I think shrinking `n_class` to 11 naively won't work, because if so, the network will output 11 classes while the ground truth still contains the 12nd class, this will lead to an unexpected error in cross entropy. And in my opinion you'd better set an `ignore_index` to exclude the unlabeled "void" class, which can be refered to in the loader of CityScapes dataset as well.

**Comment by guanfuchen** [comment]
same problem, and I wonder whether the color map is not same with the origin paper. I try to test like blew:
```python
label_colours = np.array([Sky, Building, Pole, Road, Pavement, Tree, 
                              SignSymbol, Road_marking, Car, 
                              Pedestrian, Fence, Unlabelled])
```

**Comment by komms** [comment]
@andyhahaha @d-li14  did you implement **class balancing** in the CamVid? How did you deal with **unlabelled class**?
